index.js

```diff
@@ -145,6 +149,7 @@ class NodeTestHelper extends EventEmitter {
                 Object.defineProperty(red, prop, propDescriptor);
             });
 
+        red.settings = testSettings || {};
```

Comment:
This line is whacking the `red.settings` object set in the loop above.  I think it should be:
`Object.assign(red.settings, testSettings || {});`

With this change, core node tests pass.  I suggest reverting the _user settings_ feature for now?